### PR TITLE
mgr/dashboard: get fresh inventory data from orchestrator

### PR DESCRIPTION
**Pull request by bk201**
Getting inventory data with `refresh` flag on. This is a
time-consuming call, but it's acceptable since auto-reload is not
enabled on inventory tables.

Changes:
- Inventory page/tab: fresh data are returned on initial loading and
  every manual refresh.
- OSD create form: get fresh inventory before creating OSDs.

Fixes: https://tracker.ceph.com/issues/44803
Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

## Screenshot
In OSD creation form, a spin is displayed when loading inventory:
osd-create-form

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by bk201** [comment]
jenkins test dashboard backend

File: src/pybind/mgr/dashboard/controllers/orchestrator.py [hunk 1]
```diff
@@ -102,10 +102,13 @@ def identify_device(self, hostname, device, duration):
 class OrchestratorInventory(RESTController):
 
     @raise_if_no_orchestrator
-    def list(self, hostname=None):
+    def list(self, hostname=None, refresh=None):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/orchestrator.py:105]: Note that having a `refresh=True` for cephadm is not sustainable as it blocks everything and you cannot cancel it. My thought is to completely drop support for  `refresh=True` in cephadm and raise an exception instead.

**Comment by bk201** [comment]
Closed, to be Improved in orchestrator.

**Comment by sebastian-philipp** [comment]
for future reference: #36447

**Comment by sebastian-philipp** [comment]
#36845

**Pull request closed without merging**